JM/CF/Scripts/1_Core/CommunityFramework/IO/CF_IO.c

```diff
@@ -0,0 +1,208 @@
+class CF_IO : Managed
```

Comment:
I understand the idea of a base stream, and it is good, but I would add one abstraction layer in between, to make sure you do not come to the idea to Read on a write stream or vice-versa.
Just CF_IO should not allow to read and write, one has to decide what they want it to be by strong typing it to something higher.

Move WriteX into inherited CF_WriteStream
Move ReadX into a CF_ReadStream

And inherit the other readers and writers from those intermediate classes.

Reply:
Doesn't solve anything by having an additional layer of abstraction, just makes it harder to re-use the same variable name in the same scope.

Reply:
This solves not knowing what interaction is allowed on a given stream. You are not supposed to first use a stream instance to read and then turn it into a writer. Those are usually two separate streams, of two separate types. Reading only on a ReadStream and Writing only on a WriteStream.

Reply:
I will add a return method for `IsRead` and `IsWrite`. 

`CF_Base16Stream` and `CF_Base64Stream` support both reading and writing.